config/config.go

```diff
@@ -1,62 +1,72 @@
 package config
 
 import (
-	"time"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
 
 	"github.com/spf13/viper"
 	"gopkg.in/yaml.v2"
 )
 
-// Provider defines a set of read-only methods for accessing the application
-// configuration params as defined in one of the config files.
-type Provider interface {
-	ConfigFileUsed() string
-	Get(key string) interface{}
-	GetBool(key string) bool
-	GetDuration(key string) time.Duration
-	GetFloat64(key string) float64
-	GetInt(key string) int
-	GetInt64(key string) int64
-	GetSizeInBytes(key string) uint
-	GetString(key string) string
-	GetStringMap(key string) map[string]interface{}
-	GetStringMapString(key string) map[string]string
-	GetStringMapStringSlice(key string) map[string][]string
-	GetStringSlice(key string) []string
-	GetTime(key string) time.Time
-	InConfig(key string) bool
-	IsSet(key string) bool
-}
-
 type ConfigStruct struct {
 	defaultConfig *viper.Viper
-	grafanaConfig *GrafanaConfig
+	contextMap    map[string]*GrafanaConfig
 }
 
 var configData *ConfigStruct
 
 // Config returns a default config providers
-func Config() Provider {
+func Config() *viper.Viper {
 	return configData.defaultConfig
 }
 
-func GetGrafanaConfig() *GrafanaConfig {
-	return configData.grafanaConfig
+func GetContext() string {
+	name := Config().GetString("context_name")
+	return name
+}
+
+func SetContext(context string) {
+	v := LoadConfigProvider("importer")
+	v.Set("context_name", context)
+	v.WriteConfig()
+}
+
+func GetContexts() []string {
+	return funk.Keys(configData.contextMap).([]string)
+}
+
+func GetGrafanaConfig(name string) *GrafanaConfig {
+	val, ok := configData.contextMap[name]
+	if ok {
+		return val
+	} else {
+		log.Error("Context is not found.  Please check your config")
+		os.Exit(1)
+	}
+
+	return nil
+}
+
+func GetDefaultGrafanaConfig() *GrafanaConfig {
+	return GetGrafanaConfig(GetContext())
 }
 
 // LoadConfigProvider returns a configured viper instance
-func LoadConfigProvider(appName string) Provider {
+func LoadConfigProvider(appName string) *viper.Viper {
 	return readViperConfig(appName)
 }
 
 func init() {
 	configData = &ConfigStruct{}
 	configData.defaultConfig = readViperConfig("importer")
-	grafana_config := configData.defaultConfig.GetStringMap("grafana")
-	grafana_yaml, _ := yaml.Marshal(grafana_config)
-	err := yaml.Unmarshal([]byte(grafana_yaml), &configData.grafanaConfig)
+	contexts := configData.defaultConfig.GetStringMap("contexts")
+	contextMaps, _ := yaml.Marshal(contexts)
```

Comment:
Check this error and call log.Fatal if err != nil.

Reply:
Done